src/cmd-build

```diff
@@ -23,9 +23,24 @@ cat >${commitmeta_input_json} <<EOF
 }
 EOF
 composejson=$(pwd)/work/compose.json
-# Build uses cached data
+# --cache-only is here since `fetch` is a separate verb.
 runcompose --cache-only --add-metadata-from-json ${commitmeta_input_json} \
            --write-composejson-to ${composejson}
+commit=$(ostree --repo=${workdir}/repo-build rev-parse "${ref}")
+version=$(ostree --repo=${workdir}/repo-build show --print-metadata-key=version ${commit} | sed -e "s,',,g")
+# Very special handling for --write-composejson-to as rpm-ostree doesn't
+# write it if the commit didn't change.
+if [ -f work/treecompose.changed ]; then
+    # Be nice to people who have older versions
+    mkdir -p ${workdir}/tmp
+    # Clean up prior versions
+    rm -f ${workdir}/tmp/compose-*.json
+    # Save this in case the image build fails
+    cp -a --reflink=auto ${composejson} ${workdir}/tmp/compose-${commit}.json
+else
+    # Grab the previous JSON
+    cp -a --reflink=auto ${workdir}/tmp/compose-${commit}.json work/compose.json
```

Comment:
on the first ever run of a repo we would never hit this line right, since `treecompose.changed` would always exist ?

Is there any chance someone selectively cleaned up `tmp/` and `${workdir}/tmp/compose-${commit}.json` no longer exists? should we fail gracefully in that case?

Reply:
> on the first ever run of a repo we would never hit this line right, since treecompose.changed would always exist ?

Right.

> Is there any chance someone selectively cleaned up tmp/ and ${workdir}/tmp/compose-${commit}.json no longer exists? should we fail gracefully in that case?

Why would someone do that?  We have the `clean` verb to handle this correctly right?